crypto/src/main/scala/org/bitcoins/crypto/CryptoRuntime.scala

```diff
@@ -136,4 +139,77 @@ trait CryptoRuntime {
   def sha256SchnorrNonce(bytes: ByteVector): Sha256Digest = {
     sha256(schnorrNonceTagBytes ++ bytes)
   }
+
+  def publicKey(privateKey: ECPrivateKey): ECPublicKey
```

Comment:
Ok so these are all required to be part of the `CryptoRuntime` interface? 

It seems we just need to start breaking this PR into smaller PRs that make necessary adjustments. 

This seems like it would be a good starting point.

Reply:
Yes. Everithing that calls libsecp or bouncycastle must be in `CryptoRuntime`.